report/id.go

```diff
@@ -29,25 +30,37 @@ const (
 
 // MakeEndpointNodeID produces an endpoint node ID from its composite parts.
 func MakeEndpointNodeID(hostID, namespaceID, address, port string) string {
-	return makeAddressID(hostID, namespaceID, address) + ScopeDelim + port
+	addressIP := net.ParseIP(address)
+	return makeAddressID(hostID, namespaceID, address, addressIP) + ScopeDelim + port
+}
+
+// MakeEndpointNodeIDB produces an endpoint node ID from its composite parts in binary, not strings.
+func MakeEndpointNodeIDB(hostID string, namespaceID uint64, addressIP net.IP, port uint16) string {
+	namespace := strconv.FormatUint(namespaceID, 10)
```

Comment:
This was originally preceded by:

```go
namespace := ""
if conn.Proc.NetNamespaceID > 0 {
```

Do you think that check can be dropped or we should keep it?

Reply:
I'm not sure where that quote comes from but I agree this should pass a blank string if `namespaceID==0`

Reply:
This is where it comes from :) https://github.com/weaveworks/scope/pull/3696/files#diff-3a1250bca0ad2155e60599f6f0434450L260

Reply:
Fixed.